### ISSUE TITLE
`evil` interfering with PDF text selection/highlighting in `pdf-view`

### ISSUE THREAD

**Issue by workcomplete** [doomemacs/doomemacs#6286]
### What did you expect to happen?

When I click and drag the mouse over text in a PDF, text should become highlighted. I should be able to do this consecutively (i.e. I can select text in one region, perform some action like annotation [or perform no action], and then select text in another region).

### What actually happened?

After selecting some text on a PDF page, when I click and drag the mouse over some other text on the same page nothing happens unless I click somewhere in the PDF a second time in between each highlight (i.e. I cannot perform consecutive text selection with the mouse, unless I click somewhere in the pdf between each text selection). 

I am not referring to having multiple regions highlighted or extending the original selection (simultaneous text selection) but rather being able to select different regions of text consecutively, without needing to take some action in-between each text selection.

### Describe your attempts to resolve the issue

After selecting some text, I checked the keybinding for `<mouse-1>` within the PDF, and it is bound to `(evil-mouse-drag-region START-EVENT)`. So, the above issue can be fixed by interactively disabling evil-mode by any of the following methods: toggling `evil-local-mode`, `C-z` or calling `turn-off-evil-mode`. Using any one of these I am able to make consecutive text selection without needing to click the PDF in between dragging my cursor over some region.

I also cannot disable evil-mode for pdf-view in my config. I have tried:

`(set-evil-initial-state! 'pdf-view-mode 'emacs)`
`(add-to-list '+evil-collection-disabled-list 'pdf-view-mode)`
`(add-to-list '+evil-collection-disabled-list 'pdf-view)`

I have also tried using `pdf-view-mode-hook` with the above interactive commands `evil-local-mode` and `turn-off-evil-mode` but it is my understanding that this is not the correct way to deactivate evil mode anyways.

### Steps to reproduce

1. Open Emacs
2. Open PDF using pdf-view
3. Click and drag mouse over some text on the PDF to select some text
4. Move mouse to another text region on the same PDF page and drag mouse over some text
5. Disable evil mode using `turn-off-evil-mode` or `evil-local-mode` or `C-z`
6. Repeat steps 3 and 4.

### System Information

https://pastebin.com/8EyruYQy

**Comment by asymmetric** [comment]
By the way, a related issue (also fixed by disabling `evil` mode in a `PDFView` buffer) is that after a highlight is performed (or any annotation is created, for that matter) the document is scrolled to the top of the page, and the same dance of clicking somewhere else needs to be performed before one can manually scroll back to the original point.

**Comment by 45mm** [comment]
The problem with https://github.com/syl20bnr/spacemacs/pull/15762 is that we can only yank to the kill ring, not to a named register.  Based on https://github.com/syl20bnr/spacemacs/pull/15736, I fixed the issue in another way, by adding this to my config:
```emacs-lisp
(add-hook 'pdf-view-mode-hook
          ;; disable triggering visual mode on selection in PDFView buffers
          (lambda ()
            (add-hook 'evil-local-mode-hook
                      (lambda () (remove-hook
                                  'activate-mark-hook
                                  'evil-visual-activate-hook
                                  t))
                      nil t)
            ;; implement yank ourselves
            (evil-define-key 'normal pdf-view-mode-map
              "y" #'my/pdf-view-evil-yank-visual
              )
            ))

;; based on 'pdf-view-kill-ring-save'
(defun my/pdf-view-evil-yank-visual ()
  (interactive)
  (pdf-view-assert-active-region)
  (let ((txt (pdf-view-active-region-text))
        (reg evil-this-register))
    (pdf-view-deactivate-region)
    (evil-set-register
     (or reg ?\")
     (mapconcat #'identity txt))))

```
which lets me yank to registers.

Not sure how to integrate this into Doom (or if we should), but I'm just leaving it here as one possible approach.

**Comment by tecosaur** [comment]
Reportedly fixed by 13b60b9702ea7868ecf5716d352a7e053ec6dd3f, please let us know if this isn't the case.
